src/models/model.h

```diff
@@ -59,6 +59,12 @@ struct State {
 
   std::unique_ptr<OrtRunOptions> run_options_;
 
+  // Virtual methods for prefill chunking support
+  virtual DeviceSpan<float> RunWithChunking(int total_length, DeviceSpan<int32_t>& next_tokens,
+                                            DeviceSpan<int32_t> next_indices, size_t chunk_size) {
+    return DeviceSpan<float>{};
```

Comment:
Maybe throw an exception here since this should be implemented in the derived class.